modules/command-engine/core/src/main/scala/surge/internal/kafka/KafkaProducerActorImpl.scala

```diff
@@ -406,7 +406,7 @@ class KafkaProducerActorImpl(
 
   private def publishSingleRecord(senders: Seq[ActorRef], record: ProducerRecord[String, Array[Byte]]): Future[InternalMessage] = {
     kafkaPublisherTimer.timeFuture {
-      kafkaPublisher
+      nonTransactionalStatePublisher
```

Comment:
This could get pulled out into its own change - it fixes a bug with the experimental setting for disabling transactions